pkg/controller/v1alpha2/applicationconfiguration/applicationconfiguration.go

```diff
@@ -190,6 +193,7 @@ func NewReconciler(m ctrl.Manager, o ...ReconcilerOption) *OAMApplicationReconci
 
 // Reconcile an OAM ApplicationConfigurations by rendering and instantiating its
 // Components and Traits.
+// nolint:gocyclo
```

Comment:
maybe we should keep the go-lint check and make the function clean. We can separate that logic out of the function

Reply:
I try to separate the logic of finalizers from Reconcile() func, but it's still in need of one `if err!=nil` in Reconcile func and that makes cyclomatic complexiy into 13 while gocyclo max is 12...  Is it okay for me to combine Render&Apply logic into one func and separate it from Reconcile? Or we can set the threshold of gocyclo higher,e.g. `gocyclo -over 15`?

Reply:
15 is OK I think